source/common/unified_memory_allocation/src/memory_pool.c

```diff
@@ -0,0 +1,69 @@
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ */
+
+#include <uma/memory_pool.h>
+
+#include <stdlib.h>
+
+struct uma_memory_pool_t {
+  struct uma_memory_pool_ops_t ops;
+  void *pool_priv;
+};
+
+enum uma_result_t umaPoolCreate(struct uma_memory_pool_ops_t *ops, void *params,
+                                uma_memory_pool_handle_t *hPool) {
+  uma_memory_pool_handle_t pool = malloc(sizeof(struct uma_memory_pool_t));
+  if (!pool) {
+    return UMA_RESULT_RUNTIME_ERROR;
+  }
+
+  // TODO: handle any version mismatches
+  pool->ops = *ops;
```

Comment:
I would suggest adding assert for now. Since it is the first version we do not expect a version mismatch so far. And in future, the assert will help to not forget to add required support for backward compatibility.

The same is applicable for provider ops.